URL's have changed

mossyhub opened the pull request:
there are now multiple URLs for  sending commands and the commands have changed. been building and testing this for a few days now. seems solid. cant get the poll (remote status) stuff to work though for some reason. Using Fiddler on mopar.com to build out the poll URL and params...it seems i have it all right, it does a GET on the same URL as the POST... but I gave up for now.

documentation for Home Assist for the mopar sensor needs changed for this as well.

mossyhub committed:
URL's have changed

there are now multiple URLs for  sending commands and the commands have changed. been building and testing this for a few days now. seems solid. cant get the poll (remote status) stuff to work though for some reason. Using Fiddler on mopar.com to build out the poll URL and params...it seems i have it all right, it does a GET on the same URL as the POST... but I gave up for now.

documentation for Home Assist for the mopar sensor needs changed for this as well.

mossyhub committed:
Update __init__.py

fixed a small error in the _remote_status

happyleavesaoc commented on motorparts/__init__.py:
```diff
@@ -27,18 +27,20 @@
+REMOTE_LOCK_COMMAND_URL = 'https://www.mopar.com/moparsvc/connect/lock'
+REMOTE_ENGINE_COMMAND_URL = 'https://www.mopar.com/moparsvc/connect/engine'
+REMOTE_ALARM_COMMAND_URL = 'https://www.mopar.com/moparsvc/connect/alarm'
+global REMOTE_COMMAND_URL
```
Any design should not have to use `global`. Modify `remote_command(...)` to check the `command` arg to select the correct const url, that's probably the easiest thing to do.

mossyhub replied:
I think I have it fixed now. apologize for any errors or style concerns. this is my first time using Python.

happyleavesaoc reviewed (changes requested):
Run `tox` get code style notes. Also left one comment about how to handle the varying URLs.

mossyhub committed:
updated remote_command

updated remote_command to pass REMOTE_COMMAND_URL to _remote_status as the URL is dynamically chosen per the command selected.

mossyhub committed:
fixed polling

-fixed _remote_status, all remote commands now support polling and will return a 'completed'
-a bit of cleanup

mossyhub commented on motorparts/__init__.py:
```diff
@@ -30,8 +30,6 @@
 REMOTE_LOCK_COMMAND_URL = 'https://www.mopar.com/moparsvc/connect/lock'
 REMOTE_ENGINE_COMMAND_URL = 'https://www.mopar.com/moparsvc/connect/engine'
 REMOTE_ALARM_COMMAND_URL = 'https://www.mopar.com/moparsvc/connect/alarm'
-global REMOTE_COMMAND_URL
```
removed. this was only global for the sake of passing it to _remote_status

mossyhub commented:
Anything I can do to get this merged.?I think its all good according to your comments before. I would really love to use this in Home Assistant but dont want to step on your toes and  release separately.

happyleavesaoc commented:
Hi. I merged this and then pushed another commit to update some stylistic stuff (note, you can run the style checker by `pip install tox` and then running `tox`).

Please give it a test, if it all still works, I'll publish the new version to pypi.

mossyhub commented:
works great. i had never touched python before this so honestly was just copy/pasting from your code to NotePad++ to the python IDLE as i made changes....just barely today looked into an IDE and how Tox works. Please update the mopar.py on Home Assist to match your version when you publish and this will be perfect. "OK Google...start my family wagon!"

rohankapoorcom commented:
@happyleavesaoc It looks like this was never published on pypi. Would you be able to get it published so that we can update and fix Home Assistant? See: https://github.com/home-assistant/home-assistant/issues/16379

happyleavesaoc commented:
@rohankapoorcom `1.1.0` pushed to pypi.

rohankapoorcom commented:
@happyleavesaoc upgrading Home Assistant to use 1.1.0 in https://github.com/home-assistant/home-assistant/pull/21602
